engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/TableSource.scala

```diff
@@ -86,9 +99,105 @@ class TableSource(
+
+      val result = Try {
+        tableEnv.createTable(
+          dataGenerationOutputFileInternalTableName,
+          TableDescriptor
+            .forConnector("filesystem")
+            .option("path", tempDirUrl.toString)
+            .format("json")
+            .schema(tableDefinition.toFlinkSchema)
+            .build()
+        )
+
+        val inputTable = tableEnv.from(dataGenerationInputInternalTableName)
+        inputTable.insertInto(dataGenerationOutputFileInternalTableName).execute().await()
+
+        val outputFiles: List[Path] = {
+          val dirStream: DirectoryStream[Path] = Files.newDirectoryStream(tempDir)
+          val outputFiles                      = dirStream.asScala.toList
+          dirStream.close()
+          outputFiles
+        }
+
+        val records = outputFiles
+          .flatMap(f => FileUtils.readLines(f.toFile, StandardCharsets.UTF_8).asScala)
+          .map(parse)
+          .sequence
+          .getOrElse(throw new IllegalStateException("Couldn't parse record from test data dump"))
+
+        TestData(records.map(TestRecord(_)))
+      }
+      tryToDeleteDirectoryWithLogging(tempDir)
+      result.get
+    }
+  }
+
+  private def tryToDeleteDirectoryWithLogging(dirPath: Path): Unit = Try {
+    Files
+      .walk(dirPath)
+      .sorted(java.util.Comparator.reverseOrder())
+      .forEach(path => Files.deleteIfExists(path))
+    logger.debug(s"Successfully deleted temporary test data dumping directory at: '${dirPath.toUri.toURL}'")
+  } match {
+    case Failure(e) =>
+      logger.error(
+        s"Couldn't properly delete temporary test data dumping directory at: '${dirPath.toUri.toURL}'. Exception thrown: $e"
+      )
+    case Success(_) => ()
+  }
+
+  // TODO: how to get path of jar cleaner? Through config?
+  private lazy val classPathUrlsForMiniClusterTestingEnv = List(
+    "components/flink-dev/flinkTable.jar"
+  ).map(Path.of(_).toUri.toURL)
+
+  private lazy val miniClusterTestingEnvConfig = {
+    val conf = new Configuration()
+
+    // parent-first - otherwise linkage error (loader constraint violation, a different class with the same name was
+    // previously loaded by 'app') for class 'org.apache.commons.math3.random.RandomDataGenerator'
+    conf.set(CoreOptions.CLASSLOADER_RESOLVE_ORDER, "parent-first")
+
+    // without this, on Flink taskmanager level the classloader is basically empty
+    conf.set(
+      PipelineOptions.CLASSPATHS,
+      classPathUrlsForMiniClusterTestingEnv.map(_.toString).asJava
+    )
+    conf.set(CoreOptions.DEFAULT_PARALLELISM, miniClusterTestingEnvParallelism)
+    conf
+  }
+
 }
 
 object TableSource {
   private type RECORD = java.util.Map[String, Any]
-  private val filteringInternalViewName = "filteringView"
+  private val filteringInternalViewName                 = "filteringView"
+  private val dataGenerationInputInternalTableName      = "testDataInputTable"
```

Comment:
does it mean that two Nu Designer users (working on two different batch scenarios) won't be able to use the Nu Scenario Test functionality at the same time?

Reply:
Unless there is some minicluster implementation details I don't know about, we are creating a clean environment every time the generate file functionality is called, so it shouldn't be a problem.